src/Omega_h_meshsim.cpp

```diff
@@ -108,13 +197,27 @@ void read_internal(pMesh m, Mesh* mesh) {
     }
   }
   auto ev2v = Read<LO>(host_e2v.write());
-  mesh->set_ents(Topo_type::edge, Topo_type::vertex, Adj(ev2v));
+  if (is_simplex || is_hypercube) {
+    mesh->set_ents(1, Adj(ev2v));
+    mesh->add_tag<ClassId>(1, "class_id", 1,
+                           Read<ClassId>(host_class_ids_edge.write()));
+    mesh->add_tag<I8>(1, "class_dim", 1,
+                      Read<I8>(host_class_dim_edge.write()));
+  }
+  else {
+    mesh->set_ents(Topo_type::edge, Topo_type::vertex, Adj(ev2v));
+    mesh->add_tag<ClassId>(Topo_type::edge, "class_id", 1,
+                           Read<ClassId>(host_class_ids_edge.write()));
+    mesh->add_tag<I8>(Topo_type::edge, "class_dim", 1,
+                      Read<I8>(host_class_dim_edge.write()));
+  }
+  printf("ok2\n");
```

Comment:
Please remove the debug print statements.

Reply:
Thank you for the comments. Resolved in https://github.com/SCOREC/omega_h/pull/15/commits/db4b32416f5a79e7e3c902a46c6752002b1bc880